Cross compilation with mingw for cpu=i386 from linux/macos fails with time_t bug

### Description

Compilation fails using mingw by targeting cpu=i386.

Steps to reproduce:
1. have a mingw-w64 installation for 32bit targets (v12 macos and v14 fedora tested)
2. minimal source file:
```nim
import std/times

echo getTime()
```
3. run: `nim c --cpu:i386 -d:windows -d:mingw timesbug.nim`
4. compilation fails

### Nim Version

2.0.8

### Current Output

```text
[8/8] RUN nim c --cpu:i386 -d:windows -d:mingw timesbug.nim:
0.239 Hint: used config file '/opt/nim/config/nim.cfg' [Conf]
0.239 Hint: used config file '/opt/nim/config/config.nims' [Conf]
0.243 ................................................................................................
0.710 CC: ../opt/nim/lib/system/exceptions.nim
0.711 CC: ../opt/nim/lib/std/private/since.nim
0.712 CC: ../opt/nim/lib/system/ctypes.nim
0.713 CC: ../opt/nim/lib/std/sysatomics.nim
0.714 CC: ../opt/nim/lib/system/ansi_c.nim
0.715 CC: ../opt/nim/lib/system/memory.nim
0.716 CC: ../opt/nim/lib/std/private/syslocks.nim
0.717 CC: ../opt/nim/lib/std/private/threadtypes.nim
0.720 CC: ../opt/nim/lib/std/private/digitsutils.nim
0.721 CC: ../opt/nim/lib/std/private/miscdollars.nim
0.724 CC: ../opt/nim/lib/std/assertions.nim
0.726 CC: ../opt/nim/lib/system/iterators.nim
0.729 CC: ../opt/nim/lib/system/coro_detection.nim
0.730 CC: ../opt/nim/lib/std/private/dragonbox.nim
0.731 CC: ../opt/nim/lib/std/private/schubfach.nim
0.732 CC: ../opt/nim/lib/std/formatfloat.nim
0.734 CC: ../opt/nim/lib/std/objectdollar.nim
0.737 CC: ../opt/nim/lib/system/dollars.nim
0.739 CC: ../opt/nim/lib/std/typedthreads.nim
0.740 CC: ../opt/nim/lib/system/stacktraces.nim
0.743 CC: ../opt/nim/lib/std/private/bitops_utils.nim
0.744 CC: ../opt/nim/lib/system/countbits_impl.nim
0.746 CC: ../opt/nim/lib/system/repr_v2.nim
0.747 CC: ../opt/nim/lib/std/widestrs.nim
0.749 CC: ../opt/nim/lib/core/locks.nim
0.752 CC: ../opt/nim/lib/std/exitprocs.nim
0.754 CC: ../opt/nim/lib/std/syncio.nim
0.755 CC: ../opt/nim/lib/system.nim
0.756 CC: ../opt/nim/lib/pure/parseutils.nim
0.758 CC: ../opt/nim/lib/core/macros.nim
0.761 CC: ../opt/nim/lib/pure/bitops.nim
0.764 CC: ../opt/nim/lib/pure/fenv.nim
0.765 CC: ../opt/nim/lib/pure/math.nim
0.768 CC: ../opt/nim/lib/pure/algorithm.nim
0.769 CC: ../opt/nim/lib/pure/typetraits.nim
0.771 CC: ../opt/nim/lib/std/enumutils.nim
0.773 CC: ../opt/nim/lib/std/strbasics.nim
0.774 CC: ../opt/nim/lib/pure/unicode.nim
0.776 CC: ../opt/nim/lib/std/private/jsutils.nim
0.778 CC: ../opt/nim/lib/std/private/strimpl.nim
0.780 CC: ../opt/nim/lib/pure/strutils.nim
0.781 CC: ../opt/nim/lib/pure/options.nim
0.782 CC: ../opt/nim/lib/pure/dynlib.nim
0.784 CC: ../opt/nim/lib/windows/winlean.nim
0.786 CC: ../opt/nim/lib/std/time_t.nim
0.787 CC: ../opt/nim/lib/pure/times.nim
0.790 CC: timesbug.nim
0.804 /root/.cache/nim/timesbug_d/@m..@sopt@snim@slib@spure@stimes.nim.c: In function 'getLocalOffsetAndDst__pureZtimes_u1779':
0.805 /root/.cache/nim/timesbug_d/@m..@sopt@snim@slib@spure@stimes.nim.c:3178:60: error: passing argument 1 of 'localtime' from incompatible pointer type [-Wincompatible-pointer-types]
0.805  3178 |                 nimln_(1321);           tmPtr = localtime((&a));
0.805       |                                                           ~^~~
0.805       |                                                            |
0.805       |                                                            NI32 * {aka int *}
0.805 In file included from /root/.cache/nim/timesbug_d/@m..@sopt@snim@slib@spure@stimes.nim.c:9:
0.805 /usr/i686-w64-mingw32/sys-root/mingw/include/time.h:247:62: note: expected 'const time_t *' {aka 'const long int *'} but argument is of type 'NI32 *' {aka 'int *'}
FROM fedora:40

RUN dnf install -y \
        mingw32-libgomp \
0.805   247 | static __inline struct tm *__CRTDECL localtime(const time_t *_Time) { return _localtime32(_Time); }
0.805       |                                                ~~~~~~~~~~~~~~^~~~~
0.805 /root/.cache/nim/timesbug_d/@m..@sopt@snim@slib@spure@stimes.nim.c:3213:46: error: passing argument 1 of 'localtime' from incompatible pointer type [-Wincompatible-pointer-types]
0.805  3213 |         nimln_(1330);   tmPtr_2 = localtime((&a_2));
0.805       |                                             ~^~~~~
0.805       |                                              |
0.805       |                                              NI32 * {aka int *}
0.805 /usr/i686-w64-mingw32/sys-root/mingw/include/time.h:247:62: note: expected 'const time_t *' {aka 'const long int *'} but argument is of type 'NI32 *' {aka 'int *'}
0.805   247 | static __inline struct tm *__CRTDECL localtime(const time_t *_Time) { return _localtime32(_Time); }
0.805       |                                                ~~~~~~~~~~~~~~^~~~~
0.813 Error: execution of an external compiler program '/usr/bin/i686-w64-mingw32-gcc -c  -w -fmax-errors=3 -mno-ms-bitfields -DWIN32_LEAN_AND_MEAN   -I/opt/nim/lib -I/workdir -o /root/.cache/nim/timesbug_d/@m..@sopt@snim@slib@spure@stimes.nim.c.o /root/.cache/nim/timesbug_d/@m..@sopt@snim@slib@spure@stimes.nim.c' failed with exit code: 1
0.813
0.813
------
Dockerfile:47
--------------------
  45 |         && ln -s /opt/nim/bin/nimble /usr/local/bin/nimble
  46 |
  47 | >>> RUN nim c --cpu:i386 -d:windows -d:mingw timesbug.nim
  48 |
--------------------
ERROR: failed to solve: process "/bin/sh -c nim c --cpu:i386 -d:windows -d:mingw timesbug.nim" did not complete successfully: exit code: 1
```


### Expected Output

```text
should produce executable
```


### Possible Solution

as @leorize pointed out in discord #main
> it's likely that mingw is now doing 64-bit time_t on i386 and Nim is not updated for it.

I updated std/time_t.nim for i386 branch from int32 to int64 and minimal example compiles and works fine. Not sure about other parts of stdlib which are expecting type to be 32bit.

### Additional Information

Used mingw version (docker, fedora 40):
```
i686-w64-mingw32-gcc (GCC) 14.1.1 20240607 (Fedora MinGW 14.1.1-3.fc40)
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

docker file with repro:
```
FROM fedora:40

RUN dnf install -y \
 	mingw32-libgomp \
	mingw32-bzip2-static \
	mingw32-zlib-static \
	mingw32-libpng-static \
	mingw32-libjpeg-turbo-static \
	mingw32-libtiff-static \
	mingw32-expat-static \
	mingw32-freetype-static \
	mingw32-dlfcn \
	mingw32-boost \
	mingw32-gcc-gfortran \
	mingw32-openssl-static \
	mingw32-winpthreads-static \
	git \
	sshpass \
	wine-core \
	autogen \
	unzip make patch cmake \
        xz \
	p7zip-plugins && dnf clean all

RUN ln -s /usr/bin/i686-w64-mingw32-ar /usr/local/bin/ar
RUN ln -s /usr/bin/i686-w64-mingw32-strip /usr/local/bin/strip

ENV CC i686-w64-mingw32-gcc
ENV GCC i686-w64-mingw32-gcc
ENV CXX i686-w64-mingw32-c++
ENV LANG en_US.UTF-8
ENV WINEPATH /usr/i686-w64-mingw32/sys-root/mingw/bin

# Set Nim version
ENV NIM_VERSION=2.0.8

WORKDIR /workdir

RUN echo "import std/times" >> timesbug.nim && \
    echo "echo getTime()" >> timesbug.nim
# Download and install Nim
RUN curl -L https://nim-lang.org/download/nim-${NIM_VERSION}-linux_x64.tar.xz | tar -xJ \
    && mv nim-${NIM_VERSION} /opt/nim \
    && ln -s /opt/nim/bin/nim /usr/local/bin/nim \
    && ln -s /opt/nim/bin/nimble /usr/local/bin/nimble

RUN nim c --cpu:i386 -d:windows -d:mingw timesbug.nim
```

--- a/lib/std/time_t.nim
+++ b/lib/std/time_t.nim
@@ -14,7 +14,7 @@ when defined(nimdoc):
       ## Wrapper for `time_t`. On posix, this is an alias to `posix.Time`.
 elif defined(windows):
   when defined(i386) and defined(gcc):
-    type Time* {.importc: "time_t", header: "<time.h>".} = distinct int32
+    type Time* {.importc: "time_t", header: "<time.h>".} = distinct clong
   else:
     # newest version of Visual C++ defines time_t to be of 64 bits
     type Time* {.importc: "time_t", header: "<time.h>".} = distinct int64

